tls/s2n_cipher_suites.c

```diff
@@ -131,6 +136,11 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                     continue;
                 }
 
+                /* Don't choose a suite the client shouldn't be able support */
+                if (conn->client_protocol_version < match->minimum_required_tls_version) {
+                    continue;
+                }
+
```

Comment:
"Last possible option" in most of our cipher prefs ends up being....TLS_RSA_WITH_3DES_EDE_CBC_SHA which should have no proto mismatch problems(min ver == sslv3)